f3dex2.s

```diff
@@ -547,27 +538,178 @@ OSTask:
 // RSP IMEM
 .create CODE_FILE, 0x00001080
 
-// Global registers
-secondVtxPos equ $8
-outputVtxPos equ $15
-clipFlags equ $16
-clipPolyRead equ $17
-clipPolySelect equ $18
-rdpCmdBufEnd equ $22
-rdpCmdBufPtr equ $23
-cmd_w1 equ $24
-cmd_w0 equ $25
-taskDataPtr equ $26
-inputBufferPos equ $27
-savedActiveClipPlanes equ $29
-savedRA equ $30
-
-// Global vector registers
-vZero equ $v0
-vOne equ $v1
-vPairST equ $v22
-vPairMVPPosF equ $v23
-vPairMVPPosI equ $v24
+////////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Register Use Map ///////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+// Registers marked as "global" are only used for one purpose in the vanilla
+// microcode. However, this does not necessarily mean they can't be used for
+// other things in mods--this depends on which group they're listed in below.
+
+// Note that these lists do not cover registers which are just used locally in
+// a particular region of code--you're still responsible for not breaking the
+// code you modify. This is designed to help you avoid breaking one part of the
+// code by modifying a different part.
+
+// Local register definitions are included with their code, not here.
+
+// These registers are used globally, and their values can't be rebuilt, so
+// they should never be used for anything besides their original purpose.
+//                 $zero // global
+rdpCmdBufEnd   equ $22   // global
+rdpCmdBufPtr   equ $23   // global
+taskDataPtr    equ $26   // global
+inputBufferPos equ $27   // global
+//                 $ra   // global
+
+// These registers are used throughout the codebase and expected to have
+// certain values, but you're free to overwrite them as long as you
+// reconstruct the normal values after you're done (in fact point lighting does
+// this for $v30 and $v31).
+vZero equ $v0  // global
+vOne  equ $v1  // global
+//        $v30 // global
+//        $v31 // global
+
+// Must keep values during the full clipping process: clipping overlay, vertex
+// write, tri drawing.
+clipPolySelect        equ $18 // global
+clipPolyWrite         equ $21 // also input_mtx_0
+savedActiveClipPlanes equ $29 // global
+savedRA               equ $30 // global
+
+// Must keep values during the first part of the clipping process only: polygon
+// subdivision and vertex write.
+// $2: vertex at end of edge
+clipMaskIdx  equ $5
+secondVtxPos equ $8  // global
+outputVtxPos equ $15 // global
+clipFlags    equ $16 // global
+clipPolyRead equ $17 // global
+
+// Must keep values during tri drawing.
+// They are also used throughout the codebase, but can be overwritten once their
+// use has been fulfilled for the specific command.
+cmd_w1_dram equ $24 // almost global, occasionally used locally
```

Comment:
Is there a reason that this has a `_dram` suffix while `cmd_w0` doesn't?

Reply:
Yes, it is word 1 of the command, which is also the DRAM address of whatever the command is about. Not only is it used for both purposes separately, but in most cases, it is both things at the same time. I've added a better comment to this register explaining it.